exchanges/okcoin/okcoin_websocket.go

```diff
@@ -147,584 +247,726 @@ func (o *OKCoin) WsHandleData(respRaw []byte) error {
+			Status:               status,
+			Date:                 resp.Data[x].CreateTime.Time(),
+			LastUpdated:          resp.Data[x].UpdateTime.Time(),
+			ExecutedAmount:       resp.Data[x].FillSize.Float64(),
+			ReduceOnly:           resp.Data[x].ReduceOnly,
+			Leverage:             resp.Data[x].Leverage.Float64(),
+			Price:                resp.Data[x].Price.Float64(),
+			AverageExecutedPrice: resp.Data[x].AveragePrice.Float64(),
+			RemainingAmount:      resp.Data[x].Size.Float64() - resp.Data[x].FillSize.Float64(),
+			Cost:                 resp.Data[x].AveragePrice.Float64() * resp.Data[x].FillSize.Float64(),
+			Fee:                  resp.Data[x].Fee.Float64(),
 		}
 	}
-	return trade.AddTradesToBuffer(o.Name, trades...)
+	o.Websocket.DataHandler <- algoOrder
+	return nil
 }
 
-// wsProcessCandles converts candle data and sends it to the data handler
-func (o *OKCoin) wsProcessCandles(respRaw []byte) error {
-	var response WebsocketCandleResponse
-	err := json.Unmarshal(respRaw, &response)
+func (o *Okcoin) wsProcessAccount(respRaw []byte) error {
+	var resp WebsocketAccount
+	err := json.Unmarshal(respRaw, &resp)
 	if err != nil {
 		return err
 	}
-
-	a := o.GetAssetTypeFromTableName(response.Table)
-	for i := range response.Data {
-		f := strings.Split(response.Data[i].InstrumentID, delimiterDash)
-		c := currency.NewPairWithDelimiter(f[0], f[1], delimiterDash)
-
-		timeData, err := time.Parse(time.RFC3339Nano,
-			response.Data[i].Candle[0])
-		if err != nil {
-			return fmt.Errorf("%v Time data could not be parsed: %v",
-				o.Name,
-				response.Data[i].Candle[0])
+	accountChanges := []account.Change{}
+	for a := range resp.Data {
+		for b := range resp.Data[a].Details {
+			accountChanges = append(accountChanges, account.Change{
+				Exchange: o.Name,
+				Asset:    asset.Spot,
+				Currency: currency.NewCode(resp.Data[a].Details[b].Currency),
+				Amount:   resp.Data[a].Details[b].AvailableBalance.Float64()})
 		}
+	}
+	o.Websocket.DataHandler <- accountChanges
+	return nil
+}
 
-		candleIndex := strings.LastIndex(response.Table, okcoinWsCandle)
-		candleInterval := response.Table[candleIndex+len(okcoinWsCandle):]
-
-		klineData := stream.KlineData{
-			AssetType: a,
-			Pair:      c,
-			Exchange:  o.Name,
-			Timestamp: timeData,
-			Interval:  candleInterval,
+func (o *Okcoin) wsProcessOrderbook(respRaw []byte, obChannel string) error {
+	var resp WebsocketOrderbookResponse
+	err := json.Unmarshal(respRaw, &resp)
+	if err != nil {
+		return err
+	}
+	cp, err := currency.NewPairFromString(resp.Arg.InstrumentID)
+	if err != nil {
+		return err
+	}
+	channel := resp.Arg.Channel
+	length := len(resp.Data[0].Asks) + len(resp.Data[0].Bids)
+	var snapshot bool
+	if channel == wsOrderbooks && length >= 100 ||
+		channel == wsOrderbooksL5 ||
+		channel == wsOrderbookTickByTickL50 ||
+		channel == wsOrderbookTickByTickL400 && length >= 400 ||
+		channel == wsOrderbookTickByTickL50 && length >= 50 {
+		snapshot = true
```

Comment:
Hi @samuael, shouldn't we be using resp.Action == snapshot? 

I don't think we should ignore being told that it's a snapshot, and if the orderBook is < 50 deep, we wouldn't process it correctly.

Was there a problem with using action?

Reply:
Hi @Beadko the reason why I used such conditions to classify the data as snapshot or update is the following description.
<img width="644" alt="Screenshot 2023-12-28 at 2 02 45 PM" src="https://github.com/thrasher-corp/gocryptotrader/assets/39623015/d29b04fd-20c2-4371-bf51-cad0afc0fa98">

Reply:
I am reviewing it. I'll update you when I finish with it.

Reply:
no worries, I am working in this area at the moment anyway, so don't mind taking a look at it. Just wanted to double check first :)

Reply:
Created a new PR [here](https://github.com/thrasher-corp/gocryptotrader/pull/1436)